src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,291 @@
+                ):
+                    self.job.pyprint(
+                        "{}: {}".format(self.get_ext(self.hashfile), line),
+                        file=sys.stderr,
+                    )
+                if line.endswith(self.FAILED_OPEN):
+                    self.job.pyprint(
+                        "{}: {}".format(self.get_ext(self.hashfile), line),
+                        file=sys.stderr,
+                    )
+            return err.returncode
+
+    def version(self):
+        """Return version information for the command being called."""
+        try:
+            return self._call("--version")[0]
+        except subprocess.CalledProcessError:
+            return self.COMMAND
+
+    def get_command_detail(self):
+        """Provide some way for the user to get information out of the class to
+        write METS/PREMIS provenance information.
+        """
+        if not self.command_called:
+            err_str = "Unable to retrieve information about the hashsum command called by the script"
+            raise PREMISFailure(err_str)
+        return 'program="{}"; version="{}"'.format(
+            " ".join(self.command_called), self.version()
+        )
+
+    @staticmethod
+    def _decode(out):
+        """Decode string output in Py2 or Py3 and return a list of lines to be
+        parsed elsewhere.
+        """
+        try:
+            return str(out, "utf8").split("\n")
+        except TypeError:
+            return out.decode("utf8").split("\n")
+
+    @staticmethod
+    def get_ext(path):
+        """Return the extension of the checksum file provided"""
+        ext = os.path.splitext(path)[1]
+        if not ext:
+            return path
+        return ext.replace(".", "")
+
+    @staticmethod
+    def _count_lines(path):
+        """Count the number of lines in a checksum file."""
+        count = 0
+        with open(path) as hashfile:
+            for count, _ in enumerate(hashfile):
+                pass
+        # Negate zero-based count.
+        return count + 1
+
+    @staticmethod
+    def _count_files(path):
+        """Walk the directories on a given path and count the number of files.
+        """
+        return sum([len(files) for _, _, files in os.walk(path)])
+
+
+def write_premis_event_per_file(transfer_uuid, event_detail):
+    """Write a PREMIS event for every file in the transfer on success."""
+    group_type = "transfer_id"
+    event_type = "fixity_check"
+    event_outcome = "pass"
+    # Retrieve file UUIDs from the database.
+    kwargs = {"removedtime__isnull": True, group_type: transfer_uuid}
+    with transaction.atomic():
+        file_uuids = File.objects.filter(**kwargs).values_list("uuid")
+        if not file_uuids:
+            err_str = "Unable to find the transfer objects for the SIP: '{}' in the database".format(
+                transfer_uuid
+            )
+            raise PREMISFailure(err_str)
+        for (fileUUID,) in file_uuids:
```

Comment:
I was going to suggest `file_uuids.iterator()` here ([iterator docs](https://docs.djangoproject.com/en/1.8/ref/models/querysets/#iterator)) to avoid loading all results into memory _but_ that doesn't make sense in combination with the suggestion to pass in a queryset to the function above. If we're not going to pass in a queryset that can then be cached (so with the function as is), I think iterator should improve memory usage and therefor performance here (though maybe not by much).